n3fit/src/n3fit/model_gen.py

```diff
@@ -155,56 +169,27 @@ def observable_generator(
         # Set the operation (if any) to be applied to the fktables of this dataset
         operation_name = dataset.operation
 
+        # Extract the masks that will end up in the observable wrappers...
+        trmask = mask_array[:, offset:offset + dataset.ndata] if apply_masks else None
+        tr_mask_layers.append(Mask(trmask, axis=1, c=1) if apply_masks else None)
+        vl_mask_layers.append(Mask(~trmask, axis=1, c=1) if apply_masks else None)
```

Comment:
```suggestion
        if apply_masks:
            trmask = mask_array[:, offset:offset + dataset.ndata]
            tr_mask_layers.append(Mask(bool_mask=trmask, axis=1))
            vl_mask_layers.append(Mask(bool_mask=~trmask, axis=1))
        else:
            trmask = None
            tr_mask_layers.append(None)
            vl_mask_layers.append(None)
```

I don't like the idea of having a list of `None`, I think with the check above you will always be in a consistent state and you might be able to check whether to apply the mask somewhere else (so you don't need the `None`).

Reply:
Fixed in rev. bc568761540fa4a96e91963ad314041fa81dc782